src/app.js

```diff
@@ -1,29 +1,24 @@
 import { renderApp, renderContent } from "./functions.js";
+import { generateNewObjectModal } from "./components/add-new-object/generate.new.object.js";
 
 document.addEventListener("DOMContentLoaded", () => {
-  renderApp();
-
-  document.getElementById("homeLink").addEventListener("click", () => {
-    renderContent("home");
-  });
+  document.getElementById("app").addEventListener("click", (event) => {
+    const target = event.target;
 
-  document.getElementById("classesLink").addEventListener("click", () => {
-    renderContent("class");
-  });
+    const linkClass = "nav-link";
 
-  document.getElementById("teachersLink").addEventListener("click", () => {
-    renderContent("teacher");
-  });
+    if (target.tagName === "A" && target.classList.contains(linkClass) || target.tagName === "I" ) {
```

Comment:
After I put the event listener on the add new button for only three pages,(because there is button just for these pages)I did not have a problem, but it should not have given an error for the other pages that I do not understand. Unfortunately, I get an error in the console that the event listener is not reading on the other pages.